File: C++/Count_the_Characters_V2.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

void countCharacters(string characters) {
    int size = characters.length();
    int countVowels = 0;
    int countConsonants = 0;
    int countNumber = 0;
    int countSpecial = 0;
    int whiteSpace = 0;
    int total = 0;
    char input;

    for (int counter = 0; counter < size; counter++) {
        input = characters[counter];
        if (isdigit(input)) {
            countNumber++;
        } else if (isalpha(input)) {
            input = toupper(input);
            if (input == 'A' || input == 'E' || input == 'I' || input == 'O' || input == 'U') {
                countVowels++;
            } else {
                countConsonants++;
            }
        } else if (isspace(input)) {
            whiteSpace++;
        } else {
            countSpecial++;
        }
    }

    total = countNumber + countVowels + countConsonants + whiteSpace + countSpecial;
    cout << "Number of vowels: " << countVowels << endl;
    cout << "Number of consonants: " << countConsonants << endl;
    cout << "Number of digits: " << countNumber << endl;
    cout << "Number of white spaces: " << whiteSpace << endl;
    cout << "Number of special characters: " << countSpecial << endl;
    cout << "Total number of characters: " << total << endl;
}

int main() {
    string userInput;
    cout << "Count the Characters v2" << endl;
    cout << "Enter a string: ";
    getline(cin, userInput);

    if (userInput == "") {
        cout << "No input, please input a string." << endl;
    } else {
        countCharacters(userInput);
    }

    return 0;
}

File: Python/Count_the_Characters_V2.py
def count_characters(characters):
    count_vowels = 0
    count_consonants = 0
    count_number = 0
    count_special = 0
    white_space = 0

    for ch in characters:
        if ch.isdigit():
            count_number += 1
        elif ch.isalpha():
            if ch.upper() in 'AEIOU':
                count_vowels += 1
            else:
                count_consonants += 1
        elif ch.isspace():
            white_space += 1
        else:
            count_special += 1

    total = count_number + count_vowels + count_consonants + white_space + count_special
    print("Number of vowels:", count_vowels)
    print("Number of consonants:", count_consonants)
    print("Number of digits:", count_number)
    print("Number of white spaces:", white_space)
    print("Number of special characters:", count_special)
    print("Total number of characters:", total)


print("Count the Characters v2")
user_input = input("Enter a string: ")

if user_input == "":
    print("No input, please input a string.")
else:
    count_characters(user_input)
